pkg/service/service.go

```diff
@@ -902,6 +904,22 @@ func (s *Service) DeleteService(frontend lb.L3n4Addr) (bool, error) {
 	return false, nil
 }
 
+// DeleteBackendsByIP removes the backends that match the given ip string.
+func (s *Service) DeleteBackendsByIP(ip string) error {
```

Comment:
Where is this being called? Also, why do you need a new map to keep backend entries?

Reply:
it's being called [here](https://github.com/cilium/cilium/pull/24203/files#diff-c747a60da2bed96ca522ada1e5ed841bb5c9d19a09dd17e624a9c5faf036ef33R909). 

The new map is just an optimization to quickly look up all backends to be deleted when a pod-delete event is received. The thought was that it helps avoid having to iterate through the entire list of backends to find the ones associated to the specific pod for which the delete event was received.